### PR TITLE
Ignore 504 errors while updating

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Ignroe 504 errors while updating, proxy servers throw those when they get timeout during the upgrade.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: hassio/src/dashboard/hassio-update.ts [hunk 1]
```diff
@@ -161,11 +161,18 @@ export class HassioUpdate extends LitElement {
     try {
       await this.hass.callApi<HassioResponse<void>>("POST", item.apiPath);
     } catch (err) {
-      showAlertDialog(this, {
-        title: "Update failed",
-        text:
-          typeof err === "object" ? err.body?.message || "Unkown error" : err,
-      });
+      // Only show an error if the status code was not 504 (timeout reported by proxies)
+      if (err.status_code !== 504) {
```
Comment (bramkragten) [inline hassio/src/dashboard/hassio-update.ts:165]: Return if it is 504

Comment (ludeeus) [reply hassio/src/dashboard/hassio-update.ts:165]: Why?
Not it skips the dialog, then reverts the button progress before finish (return)

Comment (bramkragten) [reply hassio/src/dashboard/hassio-update.ts:165]: Missed the last line 😄

**Review by bramkragten — APPROVED** [review]

**Comment by frenck** [comment]
Solves WTH: <https://community.home-assistant.io/t/wth-can-i-not-use-the-update-button-while-running-an-nginx-ha-proxy/221061>

**Pull request merged**